app/assets/javascripts/components/goterms.js

```diff
@@ -0,0 +1,152 @@
+import {postJSON} from "../utils.js";
+
+/**
+ * @typedef {Object} FACounts
+ * @property {number} value count
+ * @property {string} name  The name of the GO/EC number
+ * @property {string} code  The code of the GO/EC number
+ */
+
+// const for private methods
+const addData = Symbol("[addData]");
+const addMissingNames = Symbol("[addMissingNames]");
+let goData = new Map();
+
+const NAMESPACES = ["biological process", "cellular component", "molecular function"];
+
+/**
+ * Class that helps organizing GO terms
+ */
+export default class GOTerms {
+    /**
+     * Creates a new GOTerms
+     * @param  {[FACounts]} go list of GO terms with their counts
+     */
+    constructor({numAnnotatedPeptides, data}) {
+        this.numTotalSet = numAnnotatedPeptides;
+        this.go = new Map();
+        Object.values(data).forEach(v=>v.forEach(goTerm => this.go.set(goTerm.code, goTerm)));
+        GOTerms[addData](Array.from(this.go.values()));
+
+        // Sort values to store and have every namespace
+        this.data = {};
+        for (let namespace of NAMESPACES) {
+            if (namespace in data) {
+                this.data[namespace] = Array.from(data[namespace]).sort((a, b) => (b.value - a.value));
+            } else {
+                this.data[namespace] = [];
+            }
+        }
+        // Fetch names in the background, not needed yet
+        setTimeout(()=>{
+            GOTerms[addMissingNames](Array.from(this.go.keys()));
+        }, 0);
+    }
+
+    /**
+     * @return {int} number of annotated peptides
+     */
+    getTotalSetSize() {
+        return this.numTotalSet;
+    }
+
+    /**
+     * s
```

Comment:
missing text, probably due to pressing ctrl + s

_[Original comment](https://github.ugent.be/unipept/unipept/pull/625#discussion_r13721) by @bmesuere on Mon Feb 19 2018 at 14:06._